pagefaultgames/pokerogue
# [Enhancement/Item] Add Achievement to Unlock Eviolite

## What are the changes?


Adds a new achievement and unlock condition that, upon satisfying its requirement (completing Classic Mode with at least one Pokemon that can still evolve), introduces `Eviolite` into the item pool for future runs à la `Mini Black Hole`. For players, this would effectively be the official introduction of `Eviolite` as an item.

## Why am I doing these changes?




When `Eviolite` was put into the game (#2161), it was purposefully left out of the general item pool to be obtainable via other means. This PR follows up by actually introducing `Eviolite` as an unlockable item, which would address the ongoing demand for the item mentioned in its original PR.

## What did change?



- `EVIOLITE` was added to the `Unlockables` enum along with a corresponding case in `getUnlockableName` (as a fix, `Mini Black Hole` was adjusted to use its translated name in this function since it was hard-coded to English only previously). 
- The achievement, `Bring Your Child to Work Day`, was added such that its completion is validated on run clear.
- `Eviolite`'s entry in the item pool was brought back with its original weight function wrapped in an `if` checking if the item was unlocked.

### Screenshots/Videos



#### Achievement Notification

<img width="338" alt="achievementGet" src="https://github.com/user-attachments/assets/48038d3d-5b55-45f6-ae6d-e8c5bbc3f50c">


#### Achievement Menu after Completion

<img width="1215" alt="achievementScreen" src="https://github.com/user-attachments/assets/e45c1d16-c19c-47b5-bd7a-4fe3932af71e">

### Notes

- ~~With this achievement, the Achievement screen officially requires a scroll bar or icon size adjustment. I'm not well versed in UI-related changes, so this is something I'd like someone else to address either within this PR or another.~~
  - https://github.com/pagefaultgames/pokerogue/pull/3395
- On fresh data, everything fills in correctly; however, because this is a new unlock, it's relatively unclear if it'll be saved correctly since `unlocks` in `GameData` won't have an entry initializing the value for `EVIOLITE` to `false` on older saves initially. I couldn't test this situation directly after clearing my own saves, so feedback on existing cache data or in `beta` itself will be needed to make sure.

## How to test the changes?





An easy way to test whether the achievement works is by using the following override and picking any starter for a Classic Mode run:
```ts
STARTING_WAVE_OVERRIDE: 201
```
This should immediately cause Classic Mode to complete and the achievement to appear, and the text during the "Congratulations!" screen should let you know that `Eviolite` has been unlocked.

The following override can then be used in a followup run to re-roll rewards until the item appears:
```ts
WAIVE_ROLL_FEE_OVERRIDE: true
```

## Checklist
